transcom/mymove
# B-18109 I-12397 diverted shipments int

## [I-12397](https://www13.v1host.com/USTRANSCOM38/Issue.mvc/Summary?oidToken=Issue%3A892653)
## [B-18109](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A871306)

## Summary

This resolves diverted shipments attempting to calculate the weight too early.

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Is this good code practice?
- [ ] Does it resolve the issue ticket?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login as the TOO
2. Identify any HHG move that has an approved shipment
3. Request diversion on that shipment
4. Login as the prime
5. Update shipment, check the "Diversion" boolean box, apply update
6. Copy down the destination address and shipment UUID
7. Create a new HHG shipment on this move
8. Set the diversion boolean box to true on this new move
9. Set the origin address and divertedFromShipmentID to the destination address and UUID you copied, fill the rest out however you see fit
10. Create shipment
11. Login as TOO
12. Approve both shipments
13. You should not encounter this screen:
14. 
<img width="428" alt="image" src="https://github.com/transcom/mymove/assets/136814362/432603b2-7030-4d70-b5c8-ae54fa92977d">

15. Instead it should just load normally

16. 
<img width="854" alt="image" src="https://github.com/transcom/mymove/assets/136814362/bad20d39-6037-45e0-b6b6-fe32ce70c627">
